### PR TITLE
UIIN-3054 Fetch bound item related pieces from the central tenant in the central ordering mode

### PR DESCRIPTION
**Pull request by usavkov-epam**
## Purpose

When a user is on a member tenant and central ordering is enabled in the platform, it's necessary to fetch bound item-related pieces from the active tenant and the central one.

## Refs

https://folio-org.atlassian.net/browse/UIIN-3054
## Screenshots

https://github.com/user-attachments/assets/5266b308-0418-4a50-b2c5-5f34b820615d

---

non-ECS

https://github.com/user-attachments/assets/6e29d635-c20c-4268-b7af-249abd47f27d

### CODE REVIEW

File: src/common/hooks/useBoundPieces/useBoundPieces.test.js [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+describe('useBoundPieces', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+
+    useOkapiKy.mockReturnValue(mockKy);
+    useStripes.mockReturnValue(mockStripes);
+  });
+
+  it('should return default data when query is not enabled', async () => {
```
Comment (alisher-epam) [inline src/common/hooks/useBoundPieces/useBoundPieces.test.js:63]: `async` redundant

File: src/Instance/utils.js [hunk 2]
```diff
@@ -57,3 +59,15 @@ export const navigateToHoldingsViewPage = (history, location, instance, holding,
...
+  return ky.extend({
+    hooks: {
+      beforeRequest: [
+        request => {
+          request.headers.set(OKAPI_TENANT_HEADER, tenantId);
+        },
+      ],
+    },
+  });
+};
```
Comment (zburke) [inline src/Instance/utils.js:73]: What's the purpose of this? [`@folio/stripes/core::useOkapiKy()` already supports a `tenant` prop](https://github.com/folio-org/stripes-core/blob/0cdfe80aab2418acafb008650a3671babe1f0d15/src/useOkapiKy.js). Is this something different or am I misunderstanding? It's hard to tell what your goal is when there aren't any explanatory comments.

File: src/common/hooks/useBoundPieces/useBoundPieces.js [hunk 3]
```diff
@@ -29,8 +46,34 @@ const useBoundPieces = (itemId, options = {}) => {
...
+      }
+
+      return makeRequest(ky);
+    },
+    enabled: Boolean(
+      enabled
+      && itemId
+      && consortiumCentralTenantId
+      && !isCentralOrderingSettingsFetching
+    ),
```
Comment (zburke) [inline src/common/hooks/useBoundPieces/useBoundPieces.js:76]: [stripes-acq-components::getConsortiumCentralTenantId()](https://github.com/folio-org/stripes-acq-components/blob/161e6644c91066a00ca80692894c49decc198079/lib/utils/consortia/getConsortiumCentralTenantId.js#L2) will return `undefined` for non-ECS tenants. Thus, after this change, this hook will only be enabled for ECS tenants. I don't think this is correct.[](url)

**Review by zburke — COMMENTED** [review]
* I don't understand why you need ` extendKyWithTenant()` given `useOkapiKy()` will already let you override the tenant. This change looks unnecessary.
* I'm not confident `useBoundPieces()` still works correctly in non-ECS mode. This change looks incorrect.

File: src/common/hooks/useBoundPieces/useBoundPieces.js (hunk 3 above)
Comment (usavkov-epam) [reply src/common/hooks/useBoundPieces/useBoundPieces.js:76]: Good catch, thanks! Will fix it

File: src/Instance/utils.js (hunk 2 above)
Comment (usavkov-epam) [reply src/Instance/utils.js:73]: Although `useOkapiKy` supports a passed tenant, it is still a hook and imposes restrictions for hooks. We need the ability to specify the tenant for the HTTP client dynamically from the received data, in this case a react hook will not help, except as will create a basic HTTP client, which we can configure.

E.g. we have to update a piece in a tenant where it's located.
https://github.com/folio-org/ui-inventory/pull/2614/files#diff-4a37a8dce3623a54155ee5e8a9e5845b7c2f1957ef1bad4c7e919aecdd15c36fR13

File: src/common/hooks/useBoundPieces/useBoundPieces.test.js (hunk 1 above)
Comment (usavkov-epam) [reply src/common/hooks/useBoundPieces/useBoundPieces.test.js]: updated

File: src/common/hooks/useBoundPieces/useBoundPieces.js (hunk 3 above)
Comment (usavkov-epam) [reply src/common/hooks/useBoundPieces/useBoundPieces.js]: Updated

**Review by mariia-aloshyna — APPROVED** [review]

**Review by alisher-epam — APPROVED** [review]

**Pull request merged**